src/client/kube_store/index.ts

```diff
@@ -233,6 +47,48 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    let result: object = {};
+    let key: string = '';
+
+    if (resource instanceof NamespacedResource) {
+      const namespacedResource = resource as NamespacedResource;
+      if (resource instanceof CoreNamespacedResource) {
+        // Core Namespaced
+        key = `api/${resource.gvk().version}/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension Namespaced
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}` +
+          `/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    } else {
+      if (resource instanceof CoreClusterResource) {
+        // Core ClusterResource
+        key = `api/${resource.gvk().version}/namespaces/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension ClusterResource
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    }
+
+    const patchedObject = JsonMergePatch.apply(resource, patch);
+    localStorage[key] = patchedObject;
+// console.log(patchedObject)
+    return resource;
```

Comment:
We wouldn't return 'resource'.

Perhaps we'd return the patchedObject or maybe we just return a success or error condition.

Reply:
fixed